toolchain/check/cpp/macros.cpp

```diff
@@ -78,6 +79,17 @@ auto TryEvaluateMacroToConstant(Context& context, SemIR::LocId loc_id,
                                                    sema.getASTContext()));
 
   clang::APValue ap_value = evaluated_result.Val;
+  // TODO: Add support for other types.
+  if (ap_value.isLValue()) {
+    if (result_expr->EvaluateAsInt(evaluated_result, sema.getASTContext())) {
+      ap_value = evaluated_result.Val;
+    } else {
+      context.TODO(loc_id,
+                   "Unsupported: macro evaluated to a non-integer LValue");
+      return nullptr;
+    }
```

Comment:
This is more of an LLVM style suggestion than a Carbon one, so take it with a grain of salt/see how you feel about it:

```suggestion
    if (!result_expr->EvaluateAsInt(evaluated_result, sema.getASTContext())) {
      context.TODO(loc_id,
                   "Unsupported: macro evaluated to a non-integer LValue");
      return nullptr;
    }
    ap_value = evaluated_result.Val;
```

Reply:
Sg. Done.